src/Primitives.hs

```diff
@@ -695,3 +695,45 @@ primitiveDeftemplate _ ctx [s@(XObj (Sym (SymPath _ _) _) _ _), _, _, _] = do
   argumentErr ctx "deftemplate" "a symbol without prefix" "first" s
 primitiveDeftemplate _ ctx [x, _, _, _] =
   argumentErr ctx "deftemplate" "a symbol" "first" x
+
+primitiveSexpression :: Primitive
+primitiveSexpression (XObj _ i _) ctx [XObj (Sym path _) _ _] =
+  let env = contextGlobalEnv ctx
+      tyEnv = getTypeEnv $ contextTypeEnv ctx
+  in case lookupInEnv path env of
+       Just (_, binder@(Binder _ xobj)) ->
+         case xobj of
+         -- Normally, modules print their names, we actually want the deftype
+         -- form here.
+         mod@(XObj (Mod env) _ _) ->
+           case lookupInEnv path tyEnv of
+             Just (_, Binder _ (XObj (Lst forms) i t)) ->
+               return (ctx, Right (XObj (Lst (map toSymbols forms)) i t))
+             Just (_, Binder _ xobj') ->
+               return (ctx, Right xobj')
+             -- Okay, this is just a `defmodule` not a type.
+             Nothing ->
+               return (ctx, Right mod)
+         (XObj (Lst forms) i t) ->
+           return (ctx, Right (XObj (Lst (map toSymbols forms)) i t))
+         _ -> return (ctx, Right xobj)
+       Nothing ->
+         -- Just to be sure, check the type env--this might be an interface or
+         -- type
+         case lookupInEnv path tyEnv of
+           Just (_, Binder _ (XObj (Lst forms) i t)) ->
+               return (ctx, Right (XObj (Lst (map toSymbols forms)) i t))
+           Just (_, Binder _ xobj'') ->
+             return (ctx, Right xobj'')
+           Nothing -> return (ctx, dynamicNil)
```

Comment:
Should we return `()` or an error here?

Reply:
I like returning `()` instead of an error since it's quite possible a binding isn't defined and feels like a smoother experience to me to handle that case with an identity value for the purposes of these functions--that way users can decide whether and how to error out when some binding isn't defined.

`dynamicNil` is equivalent to `()` but doesn't actually emit it--how would we get the emission? Do we just return an empty `Lst` xobj here instead?

Reply:
Yep, `Lst` does it. Thanks!